backend/test/modules/email-sending/core/application/SendEmail.mailhog-spec.ts

```diff
@@ -0,0 +1,50 @@
+import { createTransport } from 'nodemailer';
+import { MailhogClient } from 'mailhog-awesome';
+
+describe('Example Jest email test-suite', () => {
+  let mailhog: MailhogClient;
+  const host = 'localhost';
+
+  beforeAll(async () => {
+    mailhog = new MailhogClient({
+      host,
+      port: 8025,
+    });
+  });
+
+  it('Retrieve the "reset password" email', async () => {
+    //GIVEN
+
+    // Inbox email only used for this test
+    const inbox = 'user1@your-domain.com';
+
+    // Clear the inbox before starting
+    await mailhog.clearInbox(inbox);
+
+    // Trigger the "reset password" email via your server's API
+    const transport = createTransport({ host, port: 1025, secure: false });
```

Comment:
Dlatego miales problem jak to zrobic w Twojej implementacji, bo Twój SendEmailCommandHandler ma za dużo odpowiedzialności, a poza tym to kilka zasad **SO**LI**D** by się tutaj przydało.. Tworzy klienta NodeMailer, zamiast dostać go z zewnątrz. Głównie wytłuszczone, ale w sumie to wszystkie :D